### PR TITLE
[4.3][GUIDEDTOURS] Cleanup Step model

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
This PR makes some minor clean up to Step model from guided tours component :

1. Simplify code of `canEditState` method. We do not have to set tour_id for the record before checking the permission. The only case we have to do that is when adding a new step, but this is already being taken care by `getItem` method
2. Simplify `save` method. We do not have get tour_id from session and set it for step. The value for that field is already passed from hidden variable on step form (and we already validate and make sure it's available when a step is being added/editted)
3. Remove some one time used variable
4. Rename variable $tourID to $tourId to follow camel case naming convention

### Testing Instructions
1. Use Joomla 4.3 nightly build or a Joomla 4.3 installation which is having Tours Guided available
2. Apply patch. Try to add/edit step for a tour, make sure nothing is broken

### Actual result BEFORE applying this Pull Request
Works

### Expected result AFTER applying this Pull Request
Works with cleaner code

### Link to documentations
Please select:
- [] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_guidedtours/src/Model/StepModel.php [hunk 1]
```diff
@@ -102,14 +102,7 @@ protected function generateNewTitle($categoryId, $alias, $title)
...
         $table      = $this->getTable();
-        $context    = $this->option . '.' . $this->name;
-        $app        = Factory::getApplication();
-        $input      = $app->input;
-        $tourID     = $app->getUserStateFromRequest($context . '.filter.tour_id', 'tour_id', 0, 'int');
-
-        if (empty($data['tour_id'])) {
-            $data['tour_id'] = $tourID;
-        }
+        $input      = Factory::getApplication()->input;
```
Comment (wilsonge) [inline administrator/components/com_guidedtours/src/Model/StepModel.php]:
```suggestion
        $input      = Factory::getApplication()->getInput();
```

**Review by wilsonge — APPROVED** [review]

File: administrator/components/com_guidedtours/src/Model/StepModel.php [hunk 2]
```diff
@@ -102,14 +102,7 @@ protected function generateNewTitle($categoryId, $alias, $title)
     public function save($data)
     {
         $table      = $this->getTable();
```
Comment (Quy) [inline administrator/components/com_guidedtours/src/Model/StepModel.php]: Please remove spaces before `=`.

**Comment by obuisard** [comment]
I have not tested this item.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40016">issues.joomla.org/tracker/joomla-cms/40016</a>.</sub>

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on efe058d0bd6cb41f397d725acbf728d1c84a2f46<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40016">issues.joomla.org/tracker/joomla-cms/40016</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40016">issues.joomla.org/tracker/joomla-cms/40016</a>.</sub>

**Comment by obuisard** [comment]
Thank you Tuan @joomdonation, appreciated!

**Pull request merged**